lib/ansible/modules/windows/win_audit_rule.ps1

```diff
@@ -0,0 +1,182 @@
+#Requires -Module Ansible.ModuleUtils.SID.psm1
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+# these are your module parameters
+$path = Get-AnsibleParam -obj $params -name "path" -type "str" -failifempty $true -aliases "destination","dest"
+$identity_reference = Get-AnsibleParam -obj $params -name "identity_reference" -type "str" -failifempty $true
+$rights = Get-AnsibleParam -obj $params -name "rights" -type "str"
+$inheritance_flags = Get-AnsibleParam -obj $params -name "inheritance_flags" -type "str" -default "ContainerInherit,ObjectInherit" -validateset 'None','ContainerInherit','ObjectInherit'
+$propagation_flags = Get-AnsibleParam -obj $params -name "propagation_flags" -type "str" -default "none" -ValidateSet 'InheritOnly','None','NoPropagateInherit'
+$audit_flags = Get-AnsibleParam -obj $params -name "audit_flags" -type "str" -default "success" -ValidateSet 'Success','Failure'
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset 'present','absent'
+
+
+<#
+
+https://msdn.microsoft.com/en-us/library/system.security.accesscontrol.inheritanceflags(v=vs.110).aspx
+https://msdn.microsoft.com/en-us/library/system.security.accesscontrol.propagationflags(v=vs.110).aspx
+
+#>
+
+#reusable get current audit rules and convert to hashtable
+Function Get-CurrentAuditRules ($path) {
+    $ACL = Get-Acl -Path $path -Audit
+
+    $HT = Foreach ($Obj in $ACL.Audit)
+    {
+        @{
+        identity_reference = $Obj.IdentityReference.ToString()
+        rights = ($Obj | select -expand "*rights").ToString()
+        audit_flags = $Obj.AuditFlags.ToString()
+        is_inherited = $Obj.IsInherited.ToString()
+        inheritance_flags = $Obj.IsInherited.ToString()
+        propagation_flags = $Obj.PropagationFlags.ToString()
+        }
+    }
+
+    $HT
+}
+
+$result = @{
+    changed = $false
+    # identity_reference = $identity_reference
+    # path = $path
+    # rights = $rights
+    # inheritance_flags = $inheritance_flags
+    # propagation_flags = $propagation_flags
+    # audit_flags = $audit_flags
+    # state = $state
+    current_audit_rules = Get-CurrentAuditRules $path
+}
+
+#Make sure target path is valid
+If (-not (Test-Path -Path $path) )
+{
+    Fail-Json -obj $result -message "defined path ($path) is not found/invalid"
+}
+
+#Make sure identity is valid and can be looked up
+Try {
+    $SID = Convert-ToSid $identity_reference
+}
+Catch {
+    Fail-Json -obj $result -message "Failed to lookup the identity ($identity_reference) - $($_.exception.message)"
+}
+
+#Make sure that the defined rights are valid and define type of audit rule
+If ($path -match "^HK(CC|CR|CU|LM|U):\\" -and $state -eq 'present')
+{
+    $PossibleRights = [System.Enum]::GetNames([System.Security.AccessControl.RegistryRights])
+    ($rights -split ',').Trim() | % {
```

Comment:
If you use the `list` type for `$rights` this can be simplified to a foreach loop without any of the splitting or trims.